Fixed problem with channels when specifying no layer name

Inuniku opened the pull request:
When specifying no layer (`layer_name.empty()`), `ChannelsInLayer `should return only layers without the '.' character.

Inuniku committed:
Fixed problem with channel order.

When specifying no layer (layer_name.empty()), ChannelsInLayer should return only layers without the '.' character.

syoyo commented:
@Inuniku Interesting. How OpenEXR handles such data? https://www.openexr.com/documentation/TechnicalIntroduction.pdf The doc says nothing when layer name is empty(e.g. ".R").

Inuniku commented:
I tweaked the change a bit with the doc saying the following:

> Layers
> ...
> Channel names that do not contain a ., or that contain a . only at the beginning or at the end are not considered to be part of any layer.

I'm citing this: https://openexr.readthedocs.io/en/latest/ReadingAndWritingImageFiles.html

Inuniku committed:
Update tinyexr.h

Accept both .LayerName and LayerName as a channel name when specifying no layer

Inuniku committed:
Update tinyexr.h

Inuniku commented:
This way, the `ChannelsInLayer` method only collects layer names without . or with . at the beginning

syoyo commented:
@Inuniku Thanks for the info! Merged!
